config/loader.js

```diff
@@ -0,0 +1,243 @@
+
+    const missingKeys = requiredKeys.map(key => {
+      const variable = process.env[key];
+      if (variable === undefined || variable === null) {
+        return key;
+      }
+    }).filter(value => value !== undefined);
+
+    // Throw an error if any required keys are missing
+    if (missingKeys.length) {
+      const message = `
+        The following required env variables are missing:
+            ${missingKeys.toString()}.
+        Please add them to your env file or run 'npm run install'
+      `;
+      throw new Error(message);
+    }
+
+    // Check JWT secret for default
+    if (process.env.JWT_SECRET === 'secret') {
+      console.warn('Warning: JWT_SECRET is set to default value');
+    }
+  }
+
+  /**
+   * Resolve the location of the env file
+   *
+   * @param {String} envFile 
+   * @returns 
+   */
+  resolve(envFile) {
+    return path.resolve(process.cwd(), envFile);
+  }
+
+  /**
+   * Add secure keys to the env
+   *
+   * @param {String} filePath The path of the .env you are updating
+   * @param {String} key The env you are adding
+   * @param {Number} length The length of the secure key
+   */
+  addSecureEnvVar(filePath, key, length) {
+    const env = {};
+    env[key] = this.generateSecureRandomString(length);
+    this.writeEnvFile(filePath, env);
+  }
+
+  /**
+   * Write the change to the env file
+   */
+  writeEnvFile(filePath, env) {
+    const content = fs.readFileSync(filePath, 'utf-8');
+    const lines = content.split('\n');
+    const updatedLines = lines.map(line => {
+      if (line.trim().startsWith('#')) {
+        // Allow comment removal
+        if (env[line] === 'remove') {
+          return null; // Mark the line for removal
+        }
+        // Preserve comments
+        return line;
+      }
+
+      const [key, value] = line.split('=');
+      if (key && value && env.hasOwnProperty(key.trim())) {
+        if (env[key.trim()] === 'remove') {
+          return null; // Mark the line for removal
+        }
+        return `${key.trim()}=${env[key.trim()]}`;
+      }
+      return line;
+    }).filter(line => line !== null); // Remove lines marked for removal
+
+    // Add any new environment variables that are not in the file yet
+    Object.entries(env).forEach(([key, value]) => {
+      if (value !== 'remove' && !updatedLines.some(line => line.startsWith(`${key}=`))) {
+        updatedLines.push(`${key}=${value}`);
+      }
+    });
+
+    // Loop through updatedLines and wrap values with spaces in double quotes
+    const fixedLines = updatedLines.map(line => {
+      const [key, value] = line.split('=');
+      if (typeof value === 'undefined' || line.trim().startsWith('#')) {
+        return line;
+      }
+      // Skip lines with quotes and numbers already
+      // Todo: this could be one regex
+      const wrappedValue = value.includes(' ') && ! value.includes('"') && ! value.includes("'") && !/\d/.test(value) ? `"${value}"` : value;
+      return `${key}=${wrappedValue}`;
```

Comment:
Ah this is another spot, this is likely a bug for the mongodb line, it'll remove everything after the = sign

Reply:
fixed